### PR TITLE
ILIAS 9 Update

### PR DESCRIPTION
**Pull request by mBeym**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/vendor/
+/.idea
+/.php-cs-fixer.cache
```
Comment (iszmais) [inline .gitignore]: Afaik we dont want any kind of cache files in our repositories, therefore those file should be excluded pre-push/deleted an ignorance is not necessary here.

`.gitingore` is mostly for required files that are not part of the versioning process.

If this is intentional, pls state you opinion on it.

Also pls add a new line EOF.

File: .php-cs-fixer.dist.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
```
Comment (iszmais) [inline .php-cs-fixer.dist.php:1]: Does this fixer differ from the one inside the ILIAS core?

If not i would **recommend** you to use the core one to promote unification of our code styles. This way you wouldn`t need to adapt your plugins fixer every time the core code style changes.

File: classes/Absences/class.xaliAbsenceStatement.php [hunk 3]
```diff
@@ -1,100 +1,109 @@
...
+ * published by ILIAS open source e-Learning e.V.
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *********************************************************************/
```
Comment (iszmais) [inline classes/Absences/class.xaliAbsenceStatement.php:17]: All license headers missing spaces (i assume this is a cs-fixer issue).

Pls use the current ILIAS 9 License header in all files.

File: classes/Absences/class.xaliAbsenceStatementFormGUI.php [hunk 4]
```diff
@@ -1,150 +1,175 @@
...
+        $this->httpWrapper = $DIC->http()->wrapper();
+
+        parent::__construct();
+
+        if ($file_id = $this->absence_statement->getFileId()) {
+            $this->ctrl->setParameter($this->parent_gui, 'file_id', $file_id);
+        }
+
+        $backCmd = $this->httpWrapper->query()->retrieve(
+            "back_cmd",
```
Comment (iszmais) [inline classes/Absences/class.xaliAbsenceStatementFormGUI.php:54]: Pls unify the quotation (here and in all other occurences).

If single or double quote i leave to your preference but it should be a unified quotation approach.

File: classes/Checklist/class.xaliChecklistGUI.php [hunk 5]
```diff
@@ -1,115 +1,135 @@
...
-			$entry = $this->checklist->getEntryOfUser($usr_id);
-			$entry->setChecklistId($this->checklist->getId());
-			$entry->setStatus($status);
-			$entry->setUserId($usr_id);
-			$entry->store();
-            if (intval($status) === xaliChecklistEntry::STATUS_ABSENT_UNEXCUSED) {
-                if (is_array($_POST['absence_reason']) && key_exists($entry->getId(), $_POST['absence_reason']) && $reason_id = $_POST['absence_reason'][$entry->getId()] !== null) {
+                $this->show();
+            }
+            */
```
Comment (iszmais) [inline classes/Checklist/class.xaliChecklistGUI.php:90]: Pls remove the ILIAS 6 Workaround completly.

File: classes/Checklist/class.xaliChecklistGUI.php [hunk 6]
```diff
@@ -1,115 +1,135 @@
...
+ * https://github.com/ILIAS-eLearning
+ *********************************************************************/
+
+declare(strict_types=1);
+
 /**
  * Class xaliChecklistGUI
  *
  * @author  Theodor Truffer <tt@studer-raimann.ch>
  */
```
Comment (iszmais) [inline classes/Checklist/class.xaliChecklistGUI.php]: Pls remove uneccessary doc blocks here and in other classes. (Mostly class and constructor docs afais)

File: classes/Overview/class.xaliOverviewGUI.php [hunk 7]
```diff
@@ -1,5 +1,19 @@
...
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *********************************************************************/
+
+declare(strict_types=1);
 
 use JetBrains\PhpStorm\NoReturn;
```
Comment (iszmais) [inline classes/Overview/class.xaliOverviewGUI.php]: Pls Remove this dependency and all `NoReturn` hints. PHP 8.1 natively Provides the return type `never` for that purpose.

File: classes/class.ilObjAttendanceListGUI.php [hunk 8]
```diff
@@ -1,359 +1,395 @@
...
+        $lng = $DIC->language();
+
+        $t = explode("_", $a_target[0]);
+        $ref_id = (int) $t[0];
+        $class_name = $a_target[1];
+
+        if (count($t) == 2) {
+            $entry_id = $t[1];
+            //			$_GET['entry_id'] = $entry_id;
+            //$ilCtrl->initBaseClass("ilObjPluginDispatchGUI");
```
Comment (iszmais) [inline classes/class.ilObjAttendanceListGUI.php]: Pls Remove all "commented" code.

**Review by iszmais — COMMENTED** [review]
LGTM. Just some minor code style issues.

Optional you might take a look into the unification of quotations to use only single quote or only double quotes.

File: .php-cs-fixer.dist.php (hunk 2 above)
Comment (mBeym) [reply .php-cs-fixer.dist.php:1]: (I) don't want plugins to be completely dependent on ILIAS.
Fixing code styling in my opinion is nothing ILIAS should interfere with

File: classes/Absences/class.xaliAbsenceStatement.php (hunk 3 above)
Comment (mBeym) [reply classes/Absences/class.xaliAbsenceStatement.php:14]: I don't see anything wrong with the license header?

File: classes/Overview/class.xaliOverviewGUI.php [hunk 9]
```diff
@@ -349,13 +476,29 @@ protected function cancel(): void
 
     #[NoReturn] protected function saveAbsenceReason(): void
```
Comment (iszmais) [inline classes/Overview/class.xaliOverviewGUI.php]: Pls use `never` here

**Review by iszmais — APPROVED** [review]
LGTM now. Just to minor miss outs.

Feel free to merge after fixing those.

**Pull request merged**